backend/src/test/repository/EntityIntegration.test.ts

```diff
@@ -664,4 +665,63 @@ describe('Integration tests for all entities', function () {
       fail('Should not have failed: ' + e);
     }
   });
+
+  test('Should be able to get all promotions that are available a specific day of the week', async () => {
+    const promotionQueryDTO: PromotionQueryDTO = {
+      dayOfWeek: Day.MONDAY,
+    };
+
+    const user1 = new UserFactory().generate();
+    const user2 = new UserFactory().generate();
+    const promotion1 = new PromotionFactory().generate(
+      user1,
+      new DiscountFactory().generate(),
+      [new ScheduleFactory().generate('8:00', '9:00', Day.MONDAY)]
+    );
+    const promotion2 = new PromotionFactory().generate(
+      user2,
+      new DiscountFactory().generate(),
+      [new ScheduleFactory().generate('8:00', '9:00', Day.TUESDAY)]
+    );
+
+    const promotion3 = new PromotionFactory().generate(
+      user2,
+      new DiscountFactory().generate(),
+      [
+        new ScheduleFactory().generate('8:00', '9:00', Day.FRIDAY),
+        new ScheduleFactory().generate('8:00', '9:00', Day.MONDAY),
+        new ScheduleFactory().generate('8:00', '9:00', Day.SATURDAY),
+      ]
+    );
+
+    await userRepository.save(user1);
+    await userRepository.save(user2);
+    await promotionRepository.save(promotion1);
+    await promotionRepository.save(promotion2);
+    await promotionRepository.save(promotion3);
+
+    try {
+      const promotions: PromotionFullTextSearch[] = await promotionRepository.getAllPromotions(
+        promotionQueryDTO
+      );
+
+      expect(promotions).toBeDefined();
+      expect(promotions.length).toEqual(2);
+      const expectedPromotions = [promotion1, promotion3];
+      for (const expectedPromotion of expectedPromotions) {
+        const p = promotions.find(
+          (promotion) => promotion.id === expectedPromotion.id
+        );
+        expect(p).toBeDefined();
+        expect(p!.schedules.length).toEqual(expectedPromotion.schedules.length);
```

Comment:
Just curious what the '!' notation means? :)

Reply:
Oh its the bang operator. Basically it tells the compiler I know it will not be undefined or null when I reach that line of code. Because on the previous line, I have an `expect` that checks that p is defined, but the compiler isn't smart enough to know that.
![image](https://user-images.githubusercontent.com/49849754/104551619-0f2a3800-55ec-11eb-9d1b-2d80a50d3235.png)